src/handlers/patient_handler.go

```diff
@@ -0,0 +1,109 @@
+package handlers
+
+import (
+	"clinic-api/src/models"
+	"clinic-api/src/utils"
+	"fmt"
+	"net/http"
+	"strings"
+
+	"github.com/google/uuid"
+	"github.com/labstack/echo/v4"
+)
+
+func CreatePatientHandler(c echo.Context) error {
+	status := http.StatusCreated
+	patientRequest := models.Patient{}
+	if err := c.Bind(&patientRequest); err != nil {
+		status = http.StatusBadRequest
+		return utils.CreateEchoResponse(c, status, nil)
+	}
+
+	newPatient := models.MapToNewPatient(patientRequest)
+	if err := models.DB.Create(&newPatient).Error; err != nil {
+		status = http.StatusInternalServerError
+		return utils.CreateEchoResponse(c, status, nil)
+	}
+
+	return utils.CreateEchoResponse(c, status, map[string]uuid.UUID{"id": newPatient.ID})
+}
+
+func GetAllPatientsHandler(c echo.Context) error {
+	status := http.StatusOK
+	name := c.QueryParam("name")
+	nik := c.QueryParam("nik")
+	patients := []models.Patient{}
+
+	query := "UPPER(name) LIKE '%" + strings.ToUpper(name) + "%'"
+	if nik != "" {
+		query = fmt.Sprint("nik = ", nik)
+	}
+
+	if err := models.DB.Where(query).Find(&patients).Error; err != nil {
+		status = http.StatusInternalServerError
+		return utils.CreateEchoResponse(c, status, nil)
+	}
+
+	return utils.CreateEchoResponse(c, status, models.MapToPatientBatch(patients))
+}
+
+func GetPatientByIDHandler(c echo.Context) error {
+	status := http.StatusOK
+	id := c.Param("id")
+	patient := models.Patient{}
+	if err := models.DB.Where("id", id).First(&patient).Error; err != nil {
+		if strings.Contains(err.Error(), "not found") {
+			status = http.StatusNotFound
+			return utils.CreateEchoResponse(c, status, nil)
+		}
+
+		status = http.StatusInternalServerError
+		return utils.CreateEchoResponse(c, status, nil)
+	}
+
+	return utils.CreateEchoResponse(c, status, models.MapToPatient(patient))
+}
+
+func EditDoctorByIDHandler(c echo.Context) error {
+	status := http.StatusNoContent
+	id := c.Param("id")
+	patientRequest := models.Patient{}
+	if err := c.Bind(&patientRequest); err != nil {
```

Comment:
If this directly bind to the model, it can be error if `dob` got from `string` type request, as created in html form. Prefer create a new struct for request then map into model.